### ISSUE TITLE
[13.0] [Wishlist] Elearning: Ability to Remove attachments on "Modify your review" on Courses

### ISSUE THREAD

**Issue by sswapnesh** [odoo/odoo#42361]
Impacted versions: 13.0

Steps to reproduce: Add Review Comments Including attachments --> Modify Review 
![Screenshot (6)](https://user-images.githubusercontent.com/16624719/71449404-c6e67080-2771-11ea-956e-e49cefd70c37.png)

Current behavior:  No Option to Remove Attachment
![Screenshot (9)](https://user-images.githubusercontent.com/16624719/71449410-d796e680-2771-11ea-8131-848104a6d077.png)

Expected behavior: Option to Remove Attachment

Video/Screenshot link (optional):

**Comment by Yenthe666** [comment]
Yep, I like the idea. @tde-banana-odoo for your interest.

**Comment by tde-banana-odoo** [comment]
Isn't it done in master ?

**Comment by sswapnesh** [comment]
Haven't checked on master yet

**Comment by pedrobaeza** [comment]
@sswapnesh did you check if this is done on master?

**Comment by sswapnesh** [comment]
Could not access Master branch

`Error: Traceback: SyntaxError: Unexpected token < in JSON at position 0 at fetch.then.res (https://2452178-master-dc3eaa.runbot31.odoo.com/web:33:142) at <anonymous>`

![Screenshot from 2020-04-30 23-13-41](https://user-images.githubusercontent.com/16624719/80741998-4200ce80-8b38-11ea-84f5-80d82c734695.png)

**Comment by pedrobaeza** [comment]
OK, when possible

**Comment by sswapnesh** [comment]
> Isn't it done in master ?

Nope, Same Behavior in master.

**Comment by tde-banana-odoo** [comment]
I am sure I implemented it one day. I will check it :) .

**Comment by tde-banana-odoo** [comment]
For ref: task 2250054 .

**Comment by tde-banana-odoo** [comment]
Hello,

Actually after investigating, it appears you can remove attachments that are not submitted, aka add attachments, see a mistake, remove them.

It is not possible to remove attachments added in the past through frontend chatter. Indeed those attachments are linked to mail.message displayed in chatter, and we keep the history. For example it could have been send to people by email. Allowing to remove them for portal users could lead to unwanted de-synchronization.

Only backend allow to remove attachments, for people having access on the document. I don't think we will change that behavior unless you have some precise use case in mind ?

Thanks,

**Comment by sswapnesh** [comment]
> unless you have some precise use case in mind ?

Nope, I don't have any but we should allow adding/remove attachment on frontend as we are allowing to change Review as well.

IMO, this behaviour should be the same for message/attachment.

**Comment by Yenthe666** [comment]
Hmm, what if the customer would like to remove a document after x time because it contains sensitive data perhaps?

**Comment by tde-banana-odoo** [comment]
You can contact the course manager if necessary.
